aztec/src/main/kotlin/org/wordpress/aztec/plugins/CssUnderlinePlugin.kt

```diff
@@ -41,7 +41,9 @@ class CssUnderlinePlugin : ISpanPostprocessor, ISpanPreprocessor {
                     if (hiddenSpan.TAG == SPAN_TAG) {
                         val parentStyle = hiddenSpan.attributes.getValue(CssStyleFormatter.STYLE_ATTRIBUTE)
                         val childStyle = calypsoUnderlineSpan.attributes.getValue(CssStyleFormatter.STYLE_ATTRIBUTE)
-                        hiddenSpan.attributes.setValue(CssStyleFormatter.STYLE_ATTRIBUTE, CssStyleFormatter.mergeStyleAttributes(parentStyle, childStyle))
+                        if (parentStyle != null && childStyle != null) {
```

Comment:
Is this part of the InputConnection issue, or solely part of https://github.com/wordpress-mobile/AztecEditor-Android/pull/832 and just bundled with this PR?

Reply:
oh that's correct it's part of #832 but I needed to keep it within this patch so my monkey runs wouldn't be capped by finding the issue fixed in #832 first and then stopping

Reply:
Ah, makes sense. :smile:

Reply:
Now that the PR [has been reverted](https://github.com/wordpress-mobile/AztecEditor-Android/pull/882), the guard here also got reverted. So, https://github.com/wordpress-mobile/AztecEditor-Android/issues/831 got opened again but I think it makes sense to re-address it on its own now. Can you wrangle that @mkevins ?

Reply:
My understanding is that this particular commit was included in both #832, and #834, and was accidentally reverted in #834, so we need to revert the revert. Does that match your understanding @hypest ? I've created this branch here: https://github.com/wordpress-mobile/AztecEditor-Android/compare/issue/831-css-underline-illegal-state-revert-revert, but so far I've not been successful in reproducing the issue on develop using the monkeys (`adb -s emulator-5554 shell monkey -p org.wordpress.aztec -v 1000000`). Do we have any documented steps to manually reproduce the crash?

Reply:
> so we need to revert the revert. Does that match your understanding @hypest ?

Heh, I wouldn't just call it "revert the revert" anymore as it's not clear if it would bring back other things besides the guard but, judging from the branch you shared I'd say that yeah, re-instating the guard is the goal.

Reply:
> Do we have any documented steps to manually reproduce the crash?

From what I recall, there weren't steps to reliably reproduce the crash, not sure if @mzorz has a better recollection on this.

Reply:
👋 ok been looking into this, this is how it goes: the reverted code made this issue reappear https://github.com/wordpress-mobile/AztecEditor-Android/issues/831, and https://github.com/wordpress-mobile/AztecEditor-Android/pull/832 has the fix we need to re-apply so, basically adding that check back will prevent the crash from happening (that means it is my understanding @mkevins 's branch has the correct code to re-apply the condition check 👍 ).
Also back then, I wasn't able to reproduce manually, but only when running the monkeys.
I'd say let's create a PR from @mkevins branch and let's apply it. Thanks for looking into this @hypest and @mkevins